### PR TITLE
ARROW-16801: [CI][C++] Use the specified MinIO instead of MinIO from Homewbrew

### PR DESCRIPTION
**Pull request by raulcd**
Our tests have problem with recent MinIO. We pin the version of MinIO to make our test stable.

See also: ARROW-16722/#13299

### CODE REVIEW

**Comment by pitrou** [comment]
Probably shouldn't bother with Homebrew and instead get a binary directly from upstream: https://dl.min.io/server/minio/release/darwin-amd64/archive/

File: .github/workflows/cpp.yml [hunk 1]
```diff
@@ -157,6 +157,7 @@ jobs:
           brew update --preinstall
           brew install --overwrite git
           brew bundle --file=cpp/Brewfile
+          /ci/scripts/install_minio.sh latest /usr/local
```
Comment (kou) [inline .github/workflows/cpp.yml:160]:
```suggestion
          ci/scripts/install_minio.sh latest /usr/local
```

**Review by kou — APPROVED** [review]
+1

**Comment by raulcd** [comment]
Many thanks for taking a look @kou! I haven't had much time in the last couple of days.

**Pull request merged**